src/Azure/Orleans.Streaming.AzureStorage/Storage/AzureQueueDataManager.cs

```diff
@@ -250,8 +251,28 @@ public async Task<QueueMessage> GetQueueMessage()
                 // http://msdn.microsoft.com/en-us/library/ee758456.aspx
                 // If no messages are visible in the queue, GetMessage returns null.
                 var client = await GetQueueClient();
-                var messages = await client.ReceiveMessagesAsync(maxMessages: 1, messageVisibilityTimeout);
-                return messages.Value.FirstOrDefault();
+                QueueProperties properties = await client.GetPropertiesAsync();
+                if (properties.ApproximateMessagesCount > 0)
+                {
+                    var messages = await client.ReceiveMessagesAsync(maxMessages: 1, messageVisibilityTimeout);
+                    var msg = messages.Value.FirstOrDefault();
+                    if (msg == null)
+                    {
+                        // Wait until the message be visible again
+                        if (messageVisibilityTimeout == null)
+                            Thread.Sleep(30000);
```

Comment:
We really should not use `Thread.Sleep` here - it will block a thread pool thread

Reply:
Thanks for you reminding @ReubenBond! Will `Task.Delay` be a better option?

Reply:
Any other feedbacks? I can modify the code if needed. 
Also, before using `ReceiveMessagesAsync`, the number of messages in the Queue is counted, which is similar to the practice here: https://docs.microsoft.com/en-us/azure/storage/queues/queues-storage-monitoring-scenarios?tabs=azure-powershell#monitor-message-counts-in-each-queue